Source/ACE/Factories/GenericObjectFactory.cs

```diff
@@ -29,37 +26,14 @@ public static List<WorldObject> CreateWorldObjects(List<AceObject> sourceObjects
                     continue;
                 }
 
-                if ((oDescFlag & ObjectDescriptionFlag.LifeStone) != 0)
-                {
-                    results.Add(new Lifestone(aceO));
-                    continue;
-                }
-                if ((oDescFlag & ObjectDescriptionFlag.Portal) != 0)
+                if (aceO.Location != null)
                 {
-                   results.Add(new Portal(aceO));
-                   continue;
-                }
-                if ((oDescFlag & ObjectDescriptionFlag.Door) != 0)
-                {
-                    results.Add(new Door(aceO));
-                    continue;
-                }
+                    WorldObject wo = new GenericObject(aceO);
 
-                switch (oType)
-                {
-#if DEBUG
-                    default:
-                        // Use the DebugObject to assist in building proper objects for weenies
-                        // FIXME(ddevec): Some objects have null location.  This freaks out the landblock... ignore them?
-                        if (aceO.Location != null)
-                        {
-                            results.Add(new DebugObject(aceO));
-                        }
-                        break;
-#endif
+                    results.Add(wo.GetObjectFromWeenieType());
```

Comment:
`var objects = DatabaseManager.World.GetObjectsByLandblock(this.id.Landblock);
            var factoryObjects = GenericObjectFactory.CreateWorldObjects(objects);
            factoryObjects.ForEach(fo =>
            {
                if (!worldObjects.ContainsKey(fo.Guid))
                {
                    worldObjects.Add(fo.Guid, fo);
                    fo.SetParent(this);
                }
            });`

GenericObjectFactory.CreateWorldObjects is called from the code above in Landblock.cs

it wants to spit out a list of WorldObjects. We need to create the objects based on their WeenieType but the function for creating shouldn't be hidden away or need to be duplicated as I believe its going to be used several times so I put it in WorldObject as GetObjectFromWeenieType. 

I don't currently see a way to get from AceObject to final real object (Lifestone, Door, etc) at this point without going through WorldObject and since that's an abstract class, I have to create something derived from it first and then use that object to create the correct weenie object.

I'm sure I'm going about this in a more convoluted method than needed, but I don't see the smarter solution presently, if you have a better way to get from AceObject to Lifestone and be able to reuse that same weenie categorizer function with generators, or spawning items, etc, we're gonna be set.